### PR TITLE
Add Using Browser Link article to client side dev

### LINKED ISSUE #2258: Write a small tutorial about how to set and use "Browser Link" with ASP.NET Core
Hi, I'd like to know if I can contribute writing a small tutorial about how to use Browser Link with ASP.NET Core.

Something like the [Using Browser Link in Visual Studio 2013](https://www.asp.net/visual-studio/overview/2013/using-browser-link) post written by Mike Wasson, with a preamble for the things needed to set it for an ASP.NET Core project:

1. Add the `Microsoft.VisualStudio.Web.BrowserLink.Loader` package with one of these options:
    - Manually editing project.json for .NET Core 1.0.x
    - Manually editing .csproj for .NET Core 1.1.0
    - Using NuGet package Manager
2. Add the `app.UseBrowserLink()` in the Startup.cs file:
```
if (env.IsDevelopment())
{
    app.UseBrowserLink();
}
```
Let me know if it can be helpful. I know it's a small contribution... :-)

### PR DESCRIPTION
**Pull request by ncarandini**
### Summary
A small tutorial about how to use Browser Link with ASP.NET Core.

Fixes #2258

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@tdykstra  please review

File: aspnetcore/client-side/using-browserlink.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+![Hover tip](using-browserlink/_static/hoover-tip.png)
+
+Change the Index view, and all connected browsers are updated when you click the Browser Link refresh button:
+
+![browsers-sync-to-changes](using-browserlink/_static/browsers-sync-to-changes.png)
+
+Browser Link also works with browsers that you launch from outside Visual Studio and navigate to the application URL.
+
+### The Browser Link Dashboard
```
Comment (tdykstra) [inline aspnetcore/client-side/using-browserlink.md]: This section was missing -- I added the title, it needs content and screenshots.

File: aspnetcore/client-side/using-browserlink.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+Open the *Startup.cs* file, and in the `Configure` method add the following code:
+
+```csharp
+app.UseBrowserLink();
+```
+
+Usually that code is inside an `if` block that enables Browser Link only in the Development environment, as shown here:
+
+![UseBrowserLink code](using-browserlink/_static/use-browserlink-code.png)
```
Comment (tdykstra) [inline aspnetcore/client-side/using-browserlink.md]: We prefer not to use images for code.  Ideally this should be a snippet that refers to a file in a sample project located in the "sample" folder.

Comment (ncarandini) [inline aspnetcore/client-side/using-browserlink.md]: Done.

File: aspnetcore/client-side/using-browserlink.md (hunk 1 above)
Comment (ncarandini) [inline aspnetcore/client-side/using-browserlink.md]: Done.

**Comment by ncarandini** [comment]
@tdykstra I did the requested changes, please verify that everything looks good.

File: aspnetcore/client-side/using-browserlink.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+![Sync with two browsers example](using-browserlink/_static/sync-with-two-browsers-example.png)
+
+Hover over the Browser Link toolbar control to see the browsers that are connected to the project:
+
+![Hover tip](using-browserlink/_static/hoover-tip.png)
+
+Change the Index view, and all connected browsers are updated when you click the Browser Link refresh button:
+
+![browsers-sync-to-changes](using-browserlink/_static/browsers-sync-to-changes.png)
```
Comment (tdykstra) [inline aspnetcore/client-side/using-browserlink.md]: missing "s" at end of "browser" in the text for number 3 ("... both browsers")

Comment (ncarandini) [inline aspnetcore/client-side/using-browserlink.md]: ops, too bad it's not my primary language, thanks for the catch!

Comment (ncarandini) [inline aspnetcore/client-side/using-browserlink.md]: Done: Image updated.

**Comment by tdykstra** [comment]
@ncarandini Are you moving this to a new PR?

**Comment by ncarandini** [comment]
Do I need to?
When I've asked this question for another small contribution I did for .NET Core, they told me it wasn't necessary because changing my fork automatically is included if you accept this PR.

But I'm still a rookie so please give me instruction for what you prefer me to do.

**Comment by tdykstra** [comment]
The PR wasn't merged yet, so it's too soon to close it and delete the branch.  Can you restore the branch and reopen the PR?

**Comment by ncarandini** [comment]
uh, I see, I've deleted the branch and merged to the main one in my fork. Silly move.
I've restored the branch and will submit a new PR, sorry for the mess.

**Comment by tdykstra** [comment]
No problem, appreciate your patience.  Before I merge it I hoped to get a review by Mads and Joe, who know more about Browser Link than I do.

**Comment by ncarandini** [comment]
Sure, and I do appreciate YOUR patience! :-)

**Comment by Rick-Anderson** [comment]
@ncarandini  I think there are a couple requested changes.

**Comment by tdykstra** [comment]
@Rick-Anderson 
Changes I requested were done by @ncarandini - we're waiting on review by @madskristensen @jodavis

**Comment by ncarandini** [comment]
@Rick-Anderson @tdykstra @madskristensen
Thanks a lot!

**Pull request merged**